### DISCUSSION TITLE
High GPU Usage

### DISCUSSION THREAD

**Discussion by teknixstuff** [Maplespe/DWMBlurGlass#186] [Q&A]
**Describe the bug**
> dwm.exe reaches very high GPU usage

**To Reproduce**
> Steps to reproduce the behavior:
> Install DWMBlurGlass
> Use default settings
> Open 5-10 windows
> Check Task Manager for highest GPU usage

**Additional information (please complete the following information):**
 > OS Version: Windows 10 21H2 [Version 10.0.19044.3803]
 > Software Version: 2.0.1

**What other third-party software do you use:**
 > StartIsBack
 > WindHawk (Aerexplorer, Bring Back the Borders!, Remove UWP titlebars, Windows 7 Alt+Tab Loader, Aero Flyout Fix, Aero Tray, Eradicate Immersive Menus, Legacy File Copy, Message Box Fix, Multi-instance regedit, No Run Icon)

**Comment by AllieTheFox** [comment]
Are you using the custom blur method ?

**Comment by teknixstuff** [comment]
> Are you using the custom blur method ?

Yes.

**Comment by digiant** [comment]
应该是StartAllBack导致的，我是AMD 5800X CPU，安装StartAllBack后，CPU温度70度，卸载后只有40度。

**Reply by teknixstuff** [reply]
I don't have startallback installed, that was a typo. I use StartIsBack++ currently.

**Comment by Maplespe** [comment]
Are you using integrated graphics + dynamic wallpaper?

**Reply by teknixstuff** [reply]
I have intergrated+dedicated graphics, and I don't have a dynamic wallpaper installed currently.

**Reply by Maplespe** [reply]
Make sure your dwm.exe is running on a discrete graphics card, in the test 5 windows in a frequently moving state the UHD 630 core graphics card had a resource usage of around 60%, while the RTX 3060 graphics card only had a resource usage of around 15% on 15+ windows.

Gaussian Blur is a GPU-intensive operation, which can lead to higher resource usage if your background is frequently updated or if you keep moving windows around.

**Reply by TorutheRedFox** [reply]
> Make sure your dwm.exe is running on a discrete graphics card, in the test 5 windows in a frequently moving state the UHD 630 core graphics card had a resource usage of around 60%, while the RTX 3060 graphics card only had a resource usage of around 15% on 15+ windows.
> 
> Gaussian Blur is a GPU-intensive operation, which can lead to higher resource usage if your background is frequently updated or if you keep moving windows around.

if only windows would actually let me do that

**Reply by berkaykrc** [reply]
How can I make sure it using dedicated GPU other than adding Graphic Settings on windows settings? Because it doesnt want to use my dedicated GPU for some reason.

**Comment by indigofairyx** [comment]
ive been getting some High GPU numbers well...

![SystemInformer_2024-03-31_12-09-10PM_642x540](https://github.com/Maplespe/DWMBlurGlass/assets/126439010/ae124628-6614-4be1-8164-ae35d69fd7c3)

![winver_2024-03-28_10-38-47AM_629x573](https://github.com/Maplespe/DWMBlurGlass/assets/126439010/8c98e4d5-94fa-48f6-aecb-52b218ee594f)

![DWMBlurGlass_2024-03-31_12-26-20PM_655x885](https://github.com/Maplespe/DWMBlurGlass/assets/126439010/f3061659-847d-4783-bc87-1906d1d13c83)

![DWMBlurGlass_2024-03-31_12-26-33PM_655x885](https://github.com/Maplespe/DWMBlurGlass/assets/126439010/7d96f8f4-5c93-4914-be0f-5beed24fe2f4)

**Comment by SurpassHR** [comment]
一样的问题，仅开机位于桌面的情况下，打开dwmblurglass待机时gpu就占用60%，关闭后5%以下；
配置: Radeon 520 + i7-10700 + win11 23H2

Same issue here: when the computer is just started and on the desktop, opening dwmblurglass causes the GPU usage to reach 60% while idle; when closed, it drops to below 5%.
System configuration: Radeon 520 + i7-10700 + win11 23H2
